subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -295,6 +298,7 @@ private
 fun compilerConfigurationFor(messageCollector: MessageCollector): CompilerConfiguration =
     CompilerConfiguration().apply {
         put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, messageCollector)
+        put(JVMConfigurationKeys.USE_FAST_CLASS_FILES_READING, true)
```

Comment:
❓ I think it is the default already, is it?

Reply:
It doesn't seem to be the default value for the code path we use. I can see code setting it to `true` based on the command line arguments but not when control flows from the API directly.  So I'd rather keep it explicitly enabled.

Reply:
Right, I mixed my mind with `-Xuse-old-class-files-reading` and the "current class files reading" being the default, see https://github.com/gradle/kotlin-dsl/pull/1004/files#diff-662c4a38f3704b9931869cb943a3c2afR40. Now it's another "fast class files reading" 🤷‍♂️ 

FTR, tests sensible to this were the ones involving parameter names metadata reading and our API jar generation.